testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m

```diff
@@ -26,10 +26,17 @@ - (void)testNoLocalePrepend {
   NSTimeInterval timeout = 10.0;
 
   // The locales received by dart:ui are exposed onBeginFrame via semantics label.
-  // There should only be one locale, since the default iOS app only has en_US as
-  // the locale. The list should consist of just the en locale.
+  // There should only be one locale. The list should consist of the default
+  // locale provided by the iOS app.
+  NSArray<NSString*>* preferredLocales = [NSLocale preferredLanguages];
+  XCTAssertEqual(preferredLocales.count, 1);
+  // Dart connects the locale parts with `_` while iOS connects them with `-`.
+  // Converts to dart format before comparing.
+  NSString* localeDart = [preferredLocales.firstObject stringByReplacingOccurrencesOfString:@"-"
+                                                                                 withString:@"_"];
+  NSString* expectedIdenntifier = [NSString stringWithFormat:@"[%@]", localeDart];
   XCUIElement* textInputSemanticsObject =
-      [self.application.textFields matchingIdentifier:@"[en]"].element;
+      [self.application.textFields matchingIdentifier:expectedIdenntifier].element;
```

Comment:
What is the value for the `expectedIdenntifier`? Curious how to differs. Also, is that difference going to be a problem is other part of our code other than just this test?

Reply:
It becomes en_US instead of en. Based on the old comment, I thought it should have always been en_US, I'm not sure why it passed before.

Reply:
When I ran with older XCode it shows en_US for me, I think the value change might be due to MacOS update rather than simulator version.